packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/CSharpGen.cs

```diff
@@ -34,37 +33,18 @@ public async Task ExecuteAsync()
             Directory.CreateDirectory(Path.Combine(generatedSourceOutputPath, "Models"));
             List<Task> generateFilesTasks = new();
 
-            foreach (TypeProvider model in output.Models)
+            foreach (var outputType in output.Types)
             {
-                generateFilesTasks.Add(workspace.AddGeneratedFile(CodeModelPlugin.Instance.GetWriter(model).Write()));
+                var writer = CodeModelPlugin.Instance.GetWriter(outputType);
+                generateFilesTasks.Add(workspace.AddGeneratedFile(writer.Write()));
 
-                foreach (var serialization in model.SerializationProviders)
+                foreach (var serialization in outputType.SerializationProviders)
                 {
-                    generateFilesTasks.Add(workspace.AddGeneratedFile(CodeModelPlugin.Instance.GetWriter(serialization).Write()));
+                    writer = CodeModelPlugin.Instance.GetWriter(serialization);
+                    generateFilesTasks.Add(workspace.AddGeneratedFile(writer.Write()));
                 }
```

Comment:
it turns out it would be more sufficiant if we do not put the serialization provider into the output type list in `OutputLibrary`, because we do not know how many we could get.
In this way, we just write them if this provider have some.

Is it possible that it is recursive? I assume it is impossible or we should avoid.